market/ipni/ipni-provider/ipni-provider.go

```diff
@@ -465,132 +482,128 @@ func (p *Provider) reconstructChunkFromDB(ctx context.Context, chunk, piece, fir
 	return b.Bytes(), nil
 }
 
+func (p *Provider) handleGetHead(w http.ResponseWriter, r *http.Request) {
+	log.Infow("Received IPNI request", "path", r.URL.Path)
+
+	providerID := chi.URLParam(r, "providerId")
```

Comment:
~~I don't think IPNI sends providerId as param in request. If it does, can we reference the source in a comment?~~
Ignore.